README.md
# Important Notice
You need atleast ElvUI version 12.00 or higher for the addon to work.

# General Information
This is a continuation of ElvUI Enhanced and a more lite weight version. Only essesential features are added and features that ElvUI can do it self are removed.

If you find any bugs or features that are not working please let me know by creating a ticket: https://github.com/nickbock/ElvUI_Enhanced_Again/issues

If you want to contribute you can find the source on github: https://github.com/nickbock/ElvUI_Enhanced_Again

# Whats New
- Moved all the setting to own settings section. This way it more easier to find settings and to change them how you want.
- Rewrite of plugin initialisation.
- Settings moved to profiles instead on a per character base.
- All features are disabled by default.

Please Note: This add-on will not function without ElvUI installed. You can download it at http://www.tukui.org.
# Features
## Mini map
- Location information above minimap with coordinates.
- Mini map buttons can be skinned and placed on their own bar that can be moved around.
## Raiding
- Raid Marker Bar for quickly placing target and world markers.
## Equipment Manager
- Automatic Equipment switching on spec change
- Automatic Equipment switching when in battleground
## Tooltip
- Player Raid Progression information on mouse-over tool tip (requires holding 'shift' key)

# Removed Features
You can make a request for ony of the old features on: github link. If enough people want it, I will consider adding the feature.
## Mini map
- Hide mini map during combat (disabled by default).
## Automation:
- Auto release in battleground after player dies.
- Automatic changing of the watched faction on your reputation bar to the faction you got reputation points for.
- Automatic role detection for battleground and party/raid groups.
- Automatic selection of highest vendor value quest rewards (you can still pick another if you like).
## Data Texts
- Target Range data text module displaying the range to your currently selected target.
- Item Level data text for a quick glance at your currently equipped item levels.
- Quick datatext configuration by CTRL-ALT-RightClick on any datatext panel.
## Equipment
- Equipment Manager with automatic changing of equipment set when changing specialization or entering a battleground / arena.
- Equipment Set Overlay for your bag and bank slots showing the equipment sets for an item.
## Visualizations
- Adjustable transparency for the ElvUI interface movers.
- Attack icon on target frame when target is not tapped by you or your group, but you still get credit for the kill when attacking.
- Improved visibility for disconnected party or raid member (lightning icon in frame).
- Item levels and durability information on the character screen.
- Watchframe visibility configuration, based on city (resting), pvp, arena, party or raid.
## Unit Frames
- GPS Locator for target and focus frame, showing direction and range to party or raid member.
- Heal Glow: Direct AoE heals will let the unit frames of the affected party / raid members glow for the defined time period.
- Hide role icons (healer/tank/damage) on unit frames while in combat.
- Class Icons for the target frame.

## ToDo
- Cleanup old files.
- Updated and cleanup locales.
- Add progression tooltip.
- Maybe add some more features.

ChangeLog
3.5.0
* Ready for 9.0
* Fix minimapbuttonbar for 9.0
* Fix raidmarkerbar for 9.0
* Fix minimaplocation for 9.0
* GPS in targetframe en focursframe for 9.0
* Thanks Cilraaz for the help!

3.4.7
* Removed support for extra data text, since you can now do this from ElvUI itself.

3.4.6
* Updated for ElvUI 11.42 (Thanks Cilraaz)

3.4.5
* Updated progression tooltip

3.4.4
* For error https://github.com/nickbock/ElvUI_Enhanced_Again/issues/100 and https://github.com/nickbock/ElvUI_Enhanced_Again/issues/99 thanks Crackpotx.

3.4.3
* Version bumb due to Curse not picking up the tag.

3.4.2
* Fix stutter https://github.com/nickbock/ElvUI_Enhanced_Again/issues/97

3.4.1
* Patch update

3.4.0
* Patch update

3.3.9
* Removed debugging info

3.3.8
* Updated for 8.2
* Updated progression for Palace and fixed BoD

3.3.7-beta1
* Fix for https://github.com/nickbock/ElvUI_Enhanced_Again/issues/94
* small change for https://github.com/nickbock/ElvUI_Enhanced_Again/issues/91

3.3.6-beta1
* Fix for ElvUI 10.88 (https://github.com/nickbock/ElvUI_Enhanced_Again/issues/91)

3.3.5-beta1
* Progression updated with Battle of Dazar'alor and Crucible of Storms
 
3.3.4-beta1
* Updated for 8.1

3.3.3-beta2
* Progression tooltip was not working

3.3.3-beta1
* Added progession tooltip for Uldir (Thanks fang2hou)

3.3.2-beta2
* Minimapbar improvements

3.3.2-beta1
* Optimalisation to remove micro stutter.
* Added functionalty for multiple columns in the minimap bar. (beta)

3.3.1-beta2
* Fix for lua error when in instance. Issue https://github.com/nickbock/ElvUI_Enhanced_Again/issues/75

3.3.1-beta1
* Added WarResources datatext based on the classhall datatext from ElvUI
* Extra fixing for garrison icon

3.3.0-beta9
* Partial Fix for https://github.com/nickbock/ElvUI_Enhanced_Again/issues/74 see issue for details.

3.3.0-beta8
* Fixed iLvl Datatext hover lua error - https://github.com/nickbock/ElvUI_Enhanced_Again/issues/50

3.3.0-beta7
* Fixed LUA error in Map Cord - https://github.com/nickbock/ElvUI_Enhanced_Again/issues/67
* Fixed Auto release in PvP - https://github.com/nickbock/ElvUI_Enhanced_Again/issues/65

3.3.0-beta6
* Fixed healglow - https://github.com/nickbock/ElvUI_Enhanced_Again/issues/62
* Fixed GPS erros - https://github.com/nickbock/ElvUI_Enhanced_Again/issues/61
* Fixed https://github.com/nickbock/ElvUI_Enhanced_Again/issues/55
* Fixed GPS module as a whole. Now only works in the open world. following Blizzards restrictions.

3.3.0-beta5
* Fixed https://github.com/nickbock/ElvUI_Enhanced_Again/issues/59

3.3.0-beta4
* Fixed https://github.com/nickbock/ElvUI_Enhanced_Again/issues/57

3.3.0-beta3
* Name change to BfA
* Fix https://github.com/nickbock/ElvUI_Enhanced_Again/issues/48

3.3.0-beta2
* Fixed lua error in minimap module

3.3.0-beta1
* Updated for Wow8
* Updated name to BtS
* Fixed minimap module
* Fixed equipment module
* Fixed bag module
* Fixed some settings that where note showing anymore
* Updated LibRangeCheck lib


3.2.4-beta2
* Fix some paperdoll itemlvl bugs.
* Progression Tooltip - Raid order changed to release order.

3.2.4-beta1
Fixed https://github.com/nickbock/ElvUI_Enhanced_Again/issues/40

3.2.3-beta2
Fixed https://github.com/nickbock/ElvUI_Enhanced_Again/issues/36
Updated progression for 7.3 (Removed EmN, added Antorus)

3.2.3-beta1
* Fix for 7.3 - changed menu option to the latest ElvUI
* Updated TOC

3.2.2-beta2
* Added Tomb of Sargeras to progression tooltip
* Fixed LUA error (https://github.com/nickbock/ElvUI_Enhanced_Again/issues/31)

3.2.2-beta1
* Updating progression tooltip with new raids
* fixed bug in calculation artifact iLVL (https://mods.curse.com/addons/wow/elvui-enhanced-again?comment=78)
* fixed missing equipmentset in equipmentset change feature
* updated API Deprecated_7_2_0 GetEquipmentSetInfo
* fixed LUA error (https://github.com/nickbock/ElvUI_Enhanced_Again/issues/28)

3.2.1-beta1
* version updated to 7.2
* removed TargetNearestUseOld feature since Blizzard reverted the tab targerting to pre-wow (by default)
* fixed bug #16 (https://github.com/nickbock/ElvUI_Enhanced_Again/issues/16)

3.2.0-beta8
* fixed hide roleicon on combat (https://github.com/nickbock/ElvUI_Enhanced_Again/issues/14)

3.2.0-beta7
* updated locationtext for wow 7.1 changes

3.2.0-beta6
* fixed lua error for elvui-10.24
* remove chatmessage for oldtabtarget

3.2.0-beta5
* Updated progression check to Legion 7.0 raids.

3.2.0-beta4
* Added an option to enable the pre legion tab targeting system. You will find this option in the general ElvUI settings. It applies the "/console TargetNearestUseOld 1" command automatically when you logon.
3.2.0-beta3
* Itemlevel for twohanded artifact weapons fixed - Thanks q3fuba for the help!
* When entering the game and a Class Hall mission is complete the alert broke the itemlvl update. This is now fixed.

3.2.0-beta2
* This update breaks support for older version of ElvUI. 
* Updated the ElvUI Enhanced config to work with the latest ElvUI update (10.17+). 
* Everyone should update to the latest ElvUI (10.19). 
* If you don't update you need to keep using 3.2.0-beta1.

3.2.0-beta1
* Changed name of the add-on and switched curseforge project
* Moved the DataText toggles to the correct settings panel
* Small localization updates
* fixed alignment bugs in the minimap enhancement (#29, #28)
* Disabled the Farmer module in preparation of deleting the module

3.1.1-beta6
* Correctly disable nameplate module (caused lua errors for some people)

3.1.1-beta5
* Updated the equipment changer to the new spec system

3.1.1-beta4
* Made the minimapbuttons fix a bit more general (tickets 24, 23)

3.1.1-beta3
* Disabled Eclipse bar libs and functions (thanks whurt74)
* Fixed small UI alignment bug in the minimap locatioin text

3.1.1-beta2

config/profile.lua
local E, L, V, P, G = unpack(ElvUI); --Inport: Engine, Locales, PrivateDB, ProfileDB, GlobalDB

--General
P['general']['minimap']['locationText'] = 'ABOVE'
P['general']['movertransparancy'] = .8

--Tooltip
P['tooltip']['progressInfo'] = true

--Unitframes
P['unitframe']['healglow'] = true
P['unitframe']['hideroleincombat'] = false
P['unitframe']['glowtime'] = .8
P['unitframe']['glowcolor'] = { r = 1, g = 1, b = 0}

P['unitframe']['units']['target']['gps'] = {
	['enable'] = true,
	['position'] = 'BOTTOMRIGHT'
}

P['unitframe']['units']['target']['attackicon'] = {
	['enable'] = true,
	['xOffset'] = 24,
	['yOffset'] = 6,
}

P['unitframe']['units']['target']['classicon'] = {
	['enable'] = true,
	['size'] = 28,
	['xOffset'] = -58,
	['yOffset'] = -22,
}

P['unitframe']['units']['focus']['gps']= {
	['enable'] = true,
	['position'] = 'LEFT'
}

-- Nameplates
P['nameplate'] = {
	['showthreat'] = true,
	['targetcount'] = true,
	['buffs'] = {
		['stretchTexture'] = false,
		['numAuras'] = 6
	},
	['debuffs'] = {
		['stretchTexture'] = false,
		['numAuras'] = 6
	},
}

-- DataTexts
P['datatexts']['Actionbar1DataPanel'] = false
P['datatexts']['Actionbar3DataPanel'] = false
P['datatexts']['Actionbar5DataPanel'] = false

P['datatexts']['panels']['Actionbar1DataPanel'] = {
	['left'] = 'Crit Chance',
	['middle'] = L['Target Range'],
	['right'] = L["Item Level"],
}

P['datatexts']['panels']['Actionbar3DataPanel'] = 'Talent/Loot Specialization'

P['datatexts']['panels']['Actionbar5DataPanel'] = 'Call to Arms'

P['watchframe'] = {
	['city'] = 'COLLAPSED',
	['pvp'] = 'HIDDEN',
	['arena'] = 'HIDDEN',
	['party'] = 'COLLAPSED',
	['raid'] = 'COLLAPSED',
}
